### DISCUSSION TITLE
How do you get a value from a column in Database and set it to a variable in Arduino program?

### DISCUSSION THREAD

**Discussion by c14nk** [tdslite/tdslite#57] [Q&A]
For example I have a column "ID" which is in table "TAB". I want to get a value from "ID" column and set it to variable "getID" in Arduino program.

Is it possible with this library and if yes, how?

**Comment by xmkg** [comment]
Hi @c14nk

Yes -- of course it's possible. That's why this library exists :)

Take a look at the [this](https://github.com/tdslite/tdslite/blob/main/examples/arduino/03-select-rows/03-select-rows.ino) sketch. In [this](https://github.com/tdslite/tdslite/blob/main/examples/arduino/03-select-rows/03-select-rows.ino#L248) line it inserts a row into `example_table` and then retrieves all rows in [this](https://github.com/tdslite/tdslite/blob/main/examples/arduino/03-select-rows/03-select-rows.ino#L260) line. The [row_callback](https://github.com/tdslite/tdslite/blob/main/examples/arduino/03-select-rows/03-select-rows.ino#L233) function would be called for each row so you can read the fields like so:

```c++
static void row_callback(void * u, const tdsl::tds_colmetadata_token & colmd,
                         const tdsl::tdsl_row & row) {
    SERIAL_PRINTLNF("row: %.4s %d", 
    // row[0] is row's first field (column `a`)
    row [0].as<tdsl::char_view>().data(),
     // row[1] is row's second field (column `b`)
                    row [1].as<tdsl::int32_t>());
}
```

Hope that helps.

**Reply by c14nk** [reply]
I'm aware of this function. My problem is I'm not sure how to implement it in the main program and how to assign the data in the "getID" variable.

**Reply by xmkg** [reply]
You need to declare a new variable and assign the field data to it:

```c++
static void row_callback(void * u, const tdsl::tds_colmetadata_token & colmd,
                         const tdsl::tdsl_row & row) {
    tdsl::uint32_t id = row[1].as<tdsl::uint32_t>();
}
```
